src/private/WRLDMapView.mm

```diff
@@ -1004,6 +1019,65 @@ - (void)notifyPoiSearchCompleted:(const Eegeo::PoiSearch::PoiSearchResults&)resu
     [self.delegate mapView:self poiSearchDidComplete:result.Id poiSearchResponse:poiSearchResponse];
 }
 
+- (void)notifyRoutingQueryCompleted:(const Eegeo::Routes::Webservice::RoutingQueryResponse&)result
+{
+    WRLDRoutingQueryResponse* routingQueryResponse = [[WRLDRoutingQueryResponse alloc] init];
+    [routingQueryResponse setSucceeded:result.Succeeded];
+    routingQueryResponse.results = [[NSMutableArray alloc] initWithCapacity:(result.Results.size())];
+    for(auto& route : result.Results)
+    {
+        WRLDRoute* wrldRoute = [[WRLDRoute alloc] init];
+        wrldRoute.sections = [[NSMutableArray alloc] initWithCapacity:(route.Sections.size())];
+        for(auto& section : route.Sections)
+        {
+            WRLDRouteSection* routeSection = [[WRLDRouteSection alloc] init];
+            routeSection.steps = [[NSMutableArray alloc] initWithCapacity:(section.Steps.size())];
+            for(auto& step : section.Steps)
+            {
+                int pathCount = static_cast<int>(step.Path.size());
+                CLLocationCoordinate2D* path = new CLLocationCoordinate2D[pathCount];
+
+                for (int i=0; i<pathCount; i++)
+                {
+                    path[i] = CLLocationCoordinate2DMake(step.Path[i].GetLatitudeInDegrees(), step.Path[i].GetLongitudeInDegrees());
+                }
+
+                const Eegeo::Routes::Webservice::RouteDirections& directions = step.Directions;
+                WRLDRouteDirections* routeDirections = [[WRLDRouteDirections alloc] init];
+                [routeDirections setType: [NSString stringWithCString: directions.Type.c_str() encoding:NSUTF8StringEncoding]];
+                [routeDirections setModifier: [NSString stringWithCString: directions.Modifier.c_str() encoding:NSUTF8StringEncoding]];
+                [routeDirections setLatLng: CLLocationCoordinate2DMake(directions.Location.GetLatitudeInDegrees(), directions.Location.GetLongitudeInDegrees())];
+                [routeDirections setHeadingBefore:directions.BearingBefore];
+                [routeDirections setHeadingAfter:directions.BearingAfter];
+
+                NSInteger mode = (NSInteger) static_cast<int>(step.Mode);
+
+                WRLDRouteStep* routeStep = [[WRLDRouteStep alloc] init];
+                [routeStep setPath:path];
```

Comment:
I've seen this pattern before with the POI service - instead of doing all the assignment via [initWith] method, setters are used instead.  Think there's a bigger discussion with wider code team on this but reckon there might be some API confusion when you receive a result object that's actually mutable?